.github/actions/pr-review/pr_review.py

```
@@ -744,6 +754,12 @@ def parse_findings(payload: object, allowed_paths: set[str], *, require_changes:
         severity = item["severity"]
         path = item["path"]
         line = item["line"]
+        # isinstance first: a JSON array or object is unhashable, and testing
+        # set membership on it raises TypeError, which is not ModelOutputError
+        # and so escapes the handler written for exactly this kind of bad
+        # model output.
+        if not isinstance(severity, str) or not isinstance(path, str):
+            raise ModelOutputError("finding has an invalid severity or path")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)